### ISSUE TITLE
Shortcuts Do Not Show Up In Filter By Default

### ISSUE THREAD

**Issue by OkyDooky** [TBog/TBLauncher#377]
# Description #
This will include the below sections, so I'll keep those short.
I just installed this launcher to try it out and got around to testing the shortcut feature by opening a page in Bromite and adding it as a shortcut. I did this to several pages, but none of them showed up in the Shortcuts section, even after switching between sections several times (Apps, Contacts, Shortcuts). I restarted my phone and nothing changed. I used the Add to Homescreen option in Mull, but nothing happened.
So, what worked was I long-pressed on one of the dock icons and went into the Edit Dock mode and added the A->Z version of the Shortcuts Filter to the dock. Then, after exiting edit mode and going to the homescreen, I tapped on the newly added A->Z version of the Shortcuts and...voila! All my shortcuts appeared in the list! And they also appeared in the regular default version of the filter.

## Context ##
Basically described above.

## Steps to Reproduce ##
To reproduce the behavior:
1. Fresh install of launcher.
2. Add shortcut through browser
3. Check default filter.
4. List is empty.

## Expected behavior ##
It should appear in the list without the need to "jostle" it, let alone in such a roundabout way.

## Device info ##
  * Device: Moto G7 Supra
  * OS: Android 9
  * TinyBit Launcher version: 7.3 F-Droid

**Comment by OkyDooky** [comment]
Update:
I removed the A->Z Shortcuts filter button from the dock and the list of shortcuts no longer shows up in regular list.

Okay, I just tested this with the Contacts, as well. Same deal. The only category that works out of the box is the Apps one. The other two defaults requires there to be a sorting version of filter in the dock, as well. 😐

Alright, just tested again, and the disappearing happens in the Apps filter, as well. And, actually, I'm not sure what causes the disappearing. I'll have to test more.

**Comment by TBog** [comment]
By default, what you see in the dock are not actions to show the apps, contacts and shortcuts.
The initial buttons are filters.

The fix for your problem is replacing the filters with actions.

**Comment by OkyDooky** [comment]
Thank you for answering.
Which, then begs the question: why have them and why make them default?
Are they simply filtering the "last used" items?

**Comment by TBog** [comment]
When toggling a filter with an empty result list all entries that the filter matches will be displayed.

**Comment by OkyDooky** [comment]
Hmmm... but, my experience seems to be that no entries get displayed. Or it will just filter the last items to populate the list...sometimes. I'll keep playing with it to try to figure out any consistencies, since this is confusing me. Even if there is no bug, its function is really hard for me to grasp.

Okay, so I think I'm figuring it out. The user needs to have the list be populated first. This can happen either by pressing an Action button, like the grid list or one for Contacts or Shortcuts, or letting it be populated by usage history automatically (what are the conditions for it to automatically show your usage history, instead of the last pressed Action button's list?). Only when it has been populated, then can the filters do their job.

Is this correct, so far?

**Comment by TBog** [comment]
> what are the conditions for it to automatically show your usage history, instead of the last pressed Action button's list?

You can customize the behavior of the launcher from TinyBit settings.

You are correct that filters only filter what is displayed in the results list. If you have only apps amd press the contact filter you'll get an empty result list.

**Comment by OkyDooky** [comment]
Thank you for being patient with me. I've been equal parts intrigued/excited by the features and frustrated/confused by the quirks and "papercuts" compared to my experience using other launchers (Lawnchair, Discreet Launcher, Lunar, Poseidon). So, I definitely want to keep trying this out and help it to get better, if I can. But, I think I might need to play with it some more before giving informed feedback.
Though, my initial feedback is that having filters as the default might be less intuitive to new users than actions, since the list would be unpopulated until the use the other features of the launcher for a while first. If there's a good counter-thought to this idea, I'm very open to it, since you obviously know more about how this launcher is designed to work than me. 😅

One potential issue I did find is that swiping up in the list does not bring up the keyboard unless there are enough entries to introduce scrolling. If you take one entry out by "remove from history" (or another list, like Shortcuts) from what's needed to scroll, swiping up no longer brings up the keyboard.
Should I open this up as a formal issue, or is this also intended?

Thanks, again.

**Comment by TBog** [comment]
It's not intended, I probably haven't tested it on short lists or it may have stopped working at some point due to other reasons.

Open a new issue so we can keep track of each issue on its own.

The initial buttons from the dock are the filters because when I started the project that was what I first implemented. I'm open to suggestions.

**Comment by OkyDooky** [comment]
>Open a new issue so we can keep track of each issue on its own.

Will do.
>The initial buttons from the dock are the filters because when I started the project that was what I first implemented. I'm open to suggestions.

Oh, okay. Obviously, this is because of my experience, but it I'd recommend having their Action versions as the starting default, for the reasons I gave above. I opened this issue because I legit thought they were broken and couldn't figure out what was wrong or how they were supposed to work. Now, after you patiently explained it to me and i had some time to actually experience it, I get what it's supposed to do and appreciate the usefulness of having filters.

Another reason I think actions would be better for a default set up is because it already includes the app button by the search bar, which basically is an action button. So, it would make sense that the other buttons would do something similar.

A different option is to include a History button (I don't yet have enough experience with the to recommend one over the other) along with the filters. Not being able to easily bring back the History list on command was part of what was confusing me (I'm still not sure when it defaults back to it automatically).

Those are the suggestions I can think of, so far.
